### PR TITLE
S2U-49: Microsoft - OneDrive integration

### PR DESCRIPTION
**Pull request by frasese**

### CODE REVIEW

File: content/content-tool/tool/src/bundle/helper_eu.properties [hunk 1]
```diff
@@ -133,3 +133,29 @@ gen.copycomf=Berretsi copyright-a
 viewing=Bistaratzen
 of=of
 items=itemetatik
+
+att.resource=Select from resources
```
Comment (mpellicer) [inline content/content-tool/tool/src/bundle/helper_eu.properties:137]: Hey @pablosanroman Could you contribute these translations please? thank you

File: microsoft-integration/README.md [hunk 2]
```diff
@@ -76,13 +76,28 @@ Mail.Send
...
+## Sakai - Microsoft Authorization Tool
+Used to manage Authorization Code Flow from Microsoft: https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow. This will enable the delegated access to Microsoft API.
+
+Requires additional delegated permissions:
+```sh
+Files.Read.All
+User.Read
+```
+
+**Important**: This tool replaces **cloud-storage/onedrive** so these deployed elements will be no longer needed:
```
Comment (mpellicer) [inline microsoft-integration/README.md:95]: This is very very important.

File: microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties [hunk 3]
```diff
@@ -128,6 +128,9 @@ config.SYNCH\:ADD_USER_TO_CHANNEL=When user is added to a group, add to related
 config.SYNCH\:REMOVE_USER_FROM_CHANNEL=When user is removed from a group, remove from related Channel(s) (requires a valid and forced Group-Channel synchronization)
 config.SYNCH\:CREATE_INVITATION=If Microsoft user does not exist, create guest invitation.
 
+config.onedrive_config=OneDrive Config
```
Comment (mpellicer) [inline microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties:131]: @Pablosanroman would you mind translating these strings? Thanks

File: microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties [hunk 4]
```diff
@@ -144,7 +147,8 @@ credentials.authority=Authority
 credentials.clientId=Client ID
 credentials.secret=Secret
 credentials.scope=Scope
-credentials.scope_alt=By default: https://graph.microsoft.com/.default
+credentials.delegated_scope = Delegated Scope
```
Comment (mpellicer) [inline microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties:150]: @Pablosanroman would you mind translating these strings? Thanks

File: microsoft-integration/pom.xml [hunk 5]
```diff
@@ -16,31 +16,22 @@
 
 	<properties>
 		<!-- Dependency versions -->
-		<msal4j.version>1.11.3</msal4j.version>
-		<azure-identity.version>1.5.0</azure-identity.version>
-		<microsoft-graph.version>5.20.0</microsoft-graph.version>
+		<azure-identity.version>1.8.3</azure-identity.version>
+		<microsoft-graph.version>5.55.0</microsoft-graph.version>
```
Comment (mpellicer) [inline microsoft-integration/pom.xml:20]: Shall we use [5.59.0](https://mvnrepository.com/artifact/com.microsoft.graph/microsoft-graph/5.59.0) ?

File: microsoft-integration/pom.xml [hunk 6]
```diff
@@ -16,31 +16,22 @@
 
 	<properties>
 		<!-- Dependency versions -->
-		<msal4j.version>1.11.3</msal4j.version>
-		<azure-identity.version>1.5.0</azure-identity.version>
-		<microsoft-graph.version>5.20.0</microsoft-graph.version>
+		<azure-identity.version>1.8.3</azure-identity.version>
```
Comment (mpellicer) [inline microsoft-integration/pom.xml:19]: Shall we use [1.9.0](https://mvnrepository.com/artifact/com.azure/azure-identity/1.9.0)?

File: microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftConfigurationServiceImpl.java [hunk 7]
```diff
@@ -64,6 +65,11 @@ public Map<String, MicrosoftConfigItem> getAllSynchronizationConfigItems(){
 		return microsoftConfigRepository.getAllSynchronizationConfigItems();
 	}
 	
+	//------------------------------ ONEDRIVE -------------------------------------------------------
+	public boolean isOneDriveEnabled() {
+		return Boolean.valueOf(microsoftConfigRepository.getConfigItemValueByKey(MicrosoftConfigRepository.ONEDRIVE_ENABLED));
```
Comment (mpellicer) [inline microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftConfigurationServiceImpl.java:70]: So, are we enabling onedrive by the admin tool? That is fantastic!

**Review by mpellicer — APPROVED** [review]
Just left some minor comments, looks amazing @frasese !

File: content/content-tool/tool/src/bundle/helper_eu.properties (hunk 1 above)
Comment (PabloSanRoman) [reply content/content-tool/tool/src/bundle/helper_eu.properties:137]: Hi @mpellicer !
att.resource=Aukeratu baliabideetatik

File: microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties (hunk 3 above)
Comment (PabloSanRoman) [reply microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties:131]: config.onedrive_config=OneDrive konfigurazioa

File: microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties (hunk 4 above)
Comment (PabloSanRoman) [reply microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties:150]: credentials.clientId=Bezeroaren IDa
credentials.secret=Sekretua
credentials.scope=Irismena
credentials.scope_alt=Lehenetsia: https://graph.microsoft.com/.default
credentials.delegated_scope = Irismen delegatua

File: content/content-tool/tool/src/bundle/helper_eu.properties (hunk 1 above)
Comment (mpellicer) [reply content/content-tool/tool/src/bundle/helper_eu.properties:137]: Thanks Pablo,updated

File: microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties (hunk 3 above)
Comment (mpellicer) [reply microsoft-integration/admin-tool/src/main/resources/Messages_eu.properties:131]: Thanks! updated

**Pull request merged**